include/envoy/http/async_client.h

```diff
@@ -170,15 +176,22 @@ class AsyncClient {
    * A structure to hold the options for AsyncRequest object.
    */
   struct RequestOptions : public StreamOptions {
-    RequestOptions() {}
-    RequestOptions(const absl::optional<std::chrono::milliseconds>& timeout)
-        : StreamOptions(timeout) {}
-    RequestOptions(const absl::optional<std::chrono::milliseconds>& timeout, bool buffer_body)
-        : StreamOptions(timeout, buffer_body) {}
-    RequestOptions(const absl::optional<std::chrono::milliseconds>& timeout, bool buffer_body,
-                   bool send_xff)
-        : StreamOptions(timeout, buffer_body, send_xff) {}
-    RequestOptions(bool buffer_body, bool send_xff) : StreamOptions(buffer_body, send_xff) {}
+    RequestOptions& setTimeout(const absl::optional<std::chrono::milliseconds>& v) {
```

Comment:
bikeshed Java nit: Is there any way to avoid repeating these definitions? Yes I know you want to return a `RequestOptions` vs. a `StreamOptions` but seems kind of bad to just copy all of this again for that reason. Is there a pattern that is typically used to deal with this?

Reply:
I know this is ugly.  But probably this is the c++ limitation. There are other options, such as template functions.  But it feel it is even worse.   

I could not come up any cleaner way than duplicating these setter functions.

Reply:
I'm leaning towards just have struct w/o setter nor constructor overload, and use multiple statement to modify them and passing in.

Reply:
I felt using setter function is more readable.  It is better than having code to specifically create a default struct and assign some values to the struct members.

So I will go with @htuch suggestion of calling Base class mutation function.